### PR TITLE
fix(VFileInput): reset internal input when model is cleared

### LINKED ISSUE #16167: [Bug Report][3.0.2] VFileInput can't select the same file after model value is set to []
### Environment
**Vuetify Version:** 3.0.2
**Vue Version:** 3.2.45
**Browsers:** Chrome 106.0.0.0
**OS:** Mac OS 10.15.7

### Steps to reproduce
Select file for VFileInput
Set VFileInput model value to `[]`  (Click "Clear" button in repro)
Select the same file for VFileInput

### Expected Behavior
VFileInput sets same file as model value

### Actual Behavior
VFileInput model value is empty. 
VFileInput looks like state should be set (label moves to the top)

### Reproduction Link
[https://play.vuetifyjs.com/#...](https://play.vuetifyjs.com/#eNp9kk1PwzAMhv+KlQsgrQ0acJkKGiAhcUbisuzQtS4rNB9K3ME07b/jpGMam+CSxn6dx2+dzDbi3rl81aOYiCJUvnV0p0yrnfUEG/DYwBYabzWccdGZMviVpBqbsu+4RBmAgNS78wveK1JUWRMImrbDALeRcD6bX8Qy4IB6b5g7qNuY3Y7iqpGWtg6TAQhQdVj6hIwRAC3bkP8wZ/MhPRyNK5MKubfPAaF2XUnIEUBRtysmliHcKqHLbKxEyrPy+sTMZ+N6glWmbY0dl6Q+SkBXLlL8gh1WhDUsuIzBtIZADFdCxm4UOQ9kYFp1bfXBB5J7bvIYv4WM4mBEshPeFfLAnxiJYd6ZLl3+Hqzhu0g/rnYCe9kPRgm+hxgrsSRyYSJlb9zHW15ZLaesSd8bajVmtdXTq3ycX99w10CH+RyDzhbefgb03FGJNMgdnNpm/VeDnczcy3w8YHepiDxBSeav0GceTY0eeST/+D6q/eX9SDvxn94BPwOxHYnhHE81mRTzb16H99A=)

### Other comments
Might be useful if the `crearable` `onClear` method was exposed, 
The comment in master/2.x VFileInput describes a similar issue: https://github.com/vuetifyjs/vuetify/blob/b0140b073ef1502d957d6234a29a0d274109c2b9/packages/vuetify/src/components/VFileInput/VFileInput.ts#L154-L159

### LINKED ISSUE #16516: [Bug Report][3.1.2] v-file-input cannot input the same file consecutively.
### Environment
**Vuetify Version:** 3.1.2
**Last working version:** 3.1.2
**Vue Version:** 3.2.45
**Browsers:** Chrome 109.0.0.0
**OS:** Windows 10

### Steps to reproduce
1. Add a file
2. Remove files programmatically
3. Add the same file
4. File not inputed (and `update:model-value` not working)

### Expected Behavior
The same file can be input

### Actual Behavior
The same file cannot input

### Reproduction Link
[https://play.vuetifyjs.com/#...](https://play.vuetifyjs.com/?preview=false#eNp9UctuwjAQ/JWVLwWJxCqlFxQQVT+DcAjJUlz8ku2kqhD/3rUTWh5tb9md2ZnJeH1kL9bmXYtszoqAysoq4LLUAEWXVdamzzSoSuhhSvNOSMyEtm0AAk2DclGyuPQlA37J3AYNq1qK+kAMo1/jV8mWDpXpEHqJdFjwRP65NRJqWXlPd1ZmMzo6QwRKsayaBqp0W3Aar8HkCIPLtg3B6F9YUSLsEXyl8F6p4EaeK6Bw3x3EoW+n4Bet0ehrJ2wAj6GNsFDWuABHyrGDE+ycUfBAdT9Ecm20H34dFpExWm/G5/VQFAGjMSyWcIy+iZt3lWyRgPWm1KcYoTclOzZhvSNFtfm7N5reNV2WA0CvM++14o6SxLlk+xCsn3Peant4y2uj+Iow7lodhMKsMWr1lE/z2TNvhA+X+xy9yrbOfHh05FiyyYV4ELvPvwwGmHQf82kvO6yi5J0UJ/0OXeZQN+jQ/Zv7hnuV/Qa7yx89qdUTdSmFPvibGmufKkzds9OE9dJETv/BNl9dGBTE)

### PR DESCRIPTION
**Pull request by floroz**
## Description

fix #16167

Currently, when the input is reset programmatically outside the VFileInput, the underlying `HTMLInputElement` is not reset, which prevents the same file from being reselected.

### What's missing from this PR

- implementing a Component testing to verify behaviour

Currently, in Cypress Component testing the Input doesn't update correctly. I might not be too familiar with the `cy` API and would require some help

## Markup:

```vue
<template>
  <v-app>
    <v-main>
      <v-file-input v-model="files" />
      <v-btn @click="onClick">remove input files</v-btn>
      <ol class="pl-4">
        <li>add a file</li>
        <li>click remove button</li>
        <li>add the same file</li>
      </ol>
    </v-main>
  </v-app>
</template>

<script setup>
import { ref } from 'vue'

const files = ref([])
const onClick = () => {
  files.value = []
}
</script>

```

## TODOS

* [ ] Add Cypress Component Test to verify correct behaviour

# Types of changes

* [x]  Bug fix (non-breaking change which fixes an issue)
* [ ]  New feature (non-breaking change which adds functionality)
* [ ]  Breaking change (fix or feature that would cause existing functionality to not work as expected)
* [ ]  Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

# Checklist:

* [x]  The PR title is no longer than 64 characters.
* [x]  The PR is submitted to the correct branch (master for bug fixes and documentation updates, dev for new features and backwards compatible changes and next for non-backwards compatible changes).
* [x]  My code follows the code style of this project.
* []  I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

File: packages/vuetify/src/components/VFileInput/__tests__/VFileInput.spec.cy.tsx [hunk 1]
```diff
@@ -136,4 +136,48 @@ describe('VFileInput', () => {
       .get('.v-file-input input')
       .should('have.attr', 'accept', 'image/*')
   })
+
+  it.only('should handle the same file to be selected via controlled model', () => {
```
Comment (floroz) [inline packages/vuetify/src/components/VFileInput/__tests__/VFileInput.spec.cy.tsx:140]: focusing until we work out how this test should look like.

Comment (nekosaur) [reply packages/vuetify/src/components/VFileInput/__tests__/VFileInput.spec.cy.tsx:140]: try using `attachFile()`

File: packages/vuetify/src/components/VFileInput/VFileInput.tsx [hunk 2]
```diff
@@ -132,6 +132,18 @@ export const VFileInput = defineComponent({
...
     }
 
+    watch(model, updatedModel => {
+      /**
+       * When model is controlled, we need to manually reset the value of the Input File Element
+       * to allow for the same file to be re-selected
+       */
+      const hasModelReset = !Array.isArray(updatedModel) || updatedModel.length === 0
+
+      if (hasModelReset && typeof inputRef.value !== 'undefined') {
```
Comment (nekosaur) [inline packages/vuetify/src/components/VFileInput/VFileInput.tsx:142]: `hasModelReset && inputRef.value` should be enough

File: packages/vuetify/src/components/VFileInput/VFileInput.tsx [hunk 3]
```diff
@@ -132,6 +132,18 @@ export const VFileInput = defineComponent({
...
+        (inputRef.value as HTMLInputElement).value = ''
```
Comment (nekosaur) [inline packages/vuetify/src/components/VFileInput/VFileInput.tsx:143]: `inputRef` is already typed as `HTMLInputElement`

**Comment by floroz** [comment]
@nekosaur Thanks for the feedback, was able to put together a good component test by using the wrapper to control the inputs :)

If you remove that watcher, the unit test will fail correctly.

File: packages/vuetify/src/components/VFileInput/__tests__/VFileInput.spec.cy.tsx [hunk 4]
```diff
@@ -136,4 +136,58 @@ describe('VFileInput', () => {
...
+      })
+      // add same file again
+      .get('.v-file-input input')
+      .attachFile('text.txt')
+      .get('.v-file-input input')
+      .should($res => {
+        const input = $res[0] as HTMLInputElement
+        expect(input.files).to.have.length(1)
+      })
+    // reset input from wrapper/parent component
```
Comment (nekosaur) [inline packages/vuetify/src/components/VFileInput/__tests__/VFileInput.spec.cy.tsx:185]: missing indentation

File: packages/vuetify/src/components/VFileInput/__tests__/VFileInput.spec.cy.tsx [hunk 5]
```diff
@@ -136,4 +136,58 @@ describe('VFileInput', () => {
       .get('.v-file-input input')
       .should('have.attr', 'accept', 'image/*')
   })
+
```
Comment (nekosaur) [inline packages/vuetify/src/components/VFileInput/__tests__/VFileInput.spec.cy.tsx:139]: add a comment with a link to the issue

File: packages/vuetify/src/components/VFileInput/__tests__/VFileInput.spec.cy.tsx [hunk 6]
```diff
@@ -136,4 +136,58 @@ describe('VFileInput', () => {
...
+      .get('button').click()
+      .get('.v-file-input input')
+      .should($res => {
+        const input = $res[0] as HTMLInputElement
+        expect(input.files).to.have.length(0)
+      })
+      // add same file again
+      .get('.v-file-input input')
+      .attachFile('text.txt')
+      .get('.v-file-input input')
```
Comment (nekosaur) [inline packages/vuetify/src/components/VFileInput/__tests__/VFileInput.spec.cy.tsx:180]: Same here.

**Review by nekosaur — CHANGES_REQUESTED** [review]

File: packages/vuetify/src/components/VFileInput/__tests__/VFileInput.spec.cy.tsx [hunk 7]
```diff
@@ -136,4 +136,58 @@ describe('VFileInput', () => {
...
+    ))
+      .get('.v-file-input input')
+      .should($res => {
+        const input = $res[0] as HTMLInputElement
+        expect(input.files).to.have.length(0)
+      })
+      .get('.v-file-input input')
+      // add file
+      .attachFile('text.txt')
+      .get('.v-file-input input')
```
Comment (nekosaur) [inline packages/vuetify/src/components/VFileInput/__tests__/VFileInput.spec.cy.tsx:165]: I think you should be able to remove all the `get`s after the first one. The context should still be the input element.

**Review by KaelWD — APPROVED** [review]

**Pull request merged**